client/service/src/lib.rs

```diff
@@ -286,10 +286,8 @@ async fn build_network_future<
 				}
 			}
 
-			// The network worker has done something. Nothing special to do, but could be
-			// used in the future to perform actions in response of things that happened on
-			// the network.
-			_ = (&mut network).fuse() => {}
+			// Drive the network. Shut down the network future if `NetworkWorker` has terminated.
+			_ = network.run().fuse() => return,
```

Comment:
Thinking about it — don't we need to break the loop if the `NetworkWorker::run()` terminated, instead of fusing it, and terminate the entire network future?

Reply:
Yeah makes sense to end the loop.

Reply:
Maybe adding some debug log when this happens.